PYD-1399: Add environments documentation (#591)
pydantic/logfire: docs/guides/advanced/using-environments.md

If you set up Logfire according to the [getting started guide](../../index.md), you already have instrumented logfire in your applications
via the `logfire.configure()` function. If you find yourself wanting to distinguish between data sent in one environment (e.g. `dev` vs. `production`), it may be overkill to create a new project for each (more on how to decide this below).

Instead you can use the environments feature, which is a special kind of resource attribute applied to the whole payload received by Logfire.
This attribute says which environment the payload comes from.

> You can set a Logfire environment via the SDK `logfire.configure` function:

Reviewer: I wouldn't call it "Logfire environment", just "environment".

```suggestion
You can set the environment for your project when calling `logfire.configure`:
```

```py title="main.py"
import logfire

logfire.configure(environment='dev')

> from app.main import main
> 
> main()

Reviewer: I prefer to have examples people can reproduce by _copy-pasting_.
```suggestion
logfire.info("Hi there!")
```

```

Under the hood, this sets the OTel [`deployment.environment.name`](https://opentelemetry.io/docs/specs/semconv/resource/deployment-environment/).

> Note that you can also set this via the `LOGFIRE_ENVIRONMENT` environment variable.

Reviewer: I think more important than `LOGFIRE_ENVIRONMENT` would be to explain how to set this in other languages... e.g. with the `OTEL_RESOURCE_ATTRIBUTES="deployment.environment.name=prod"` environment variable.

Author: Thanks, added

Once set, you will see your environment in the Logfire UI `all envs` dropdown, which appears
on the [Live View](../web-ui/live.md), [Dashboards](../web-ui/dashboards.md) and [Explore](../web-ui/explore.md) pages:

![Environments](../../images/guide/environments.png)

Note that by default there are system generated environments:

> - `all envs`: Searches will include all spans with any environment set

Author: @Kludex is this description of `all_envs` vs. `not specified` correct, I wasn't sure. Could it be that `not specified` just returns everything?

Reviewer 2: All envs has no limitations,

Reviewer 2: `not specified` returns logs with empty env value

- `not specified`: Searches will include all spans with no environment specified

> Any environments you create via the SDK will appear below the system generated envs. When you select an environment,

Reviewer: I think this bot is actually correct.

all subsequent queries (e.g. on live view, dashboards or explore) will filter by that environment.

## Can I Create an Environment in the UI?

> No, you cannot create or delete set environments via the UI, instead use the SDK.

Reviewer: I wouldn't include unnecessary information. 👀

Author: Disagree, I think it's worth being explicit about this so people don't try and find it. There is very little downside to clarifying it.

## How do I delete an environment?
Once an environment has been configured and received by logfire, technically it’s available for
the length of the data retention period while that environment exists in the data.
You can however add new ones, and change the configuration of which data is assigned to which
environment name.

## Should I Use Environments or Projects?

Environments are more lightweight than projects. Projects give you the ability to assign specific

> user groups and permissions levels (see this [organization structure diagram](docs/reference/organization-structure/) for details).

Reviewer: Can you use relative path to the docs like the other links, please?

Author: fixed

> If you set up **Logfire** according to the [getting started guide](../../index.md), you already
> have instrumented your applications via the `logfire.configure()` function.
> 
> If you find yourself wanting to distinguish between data sent in one environment (e.g. `dev` vs. `production`),
> it may be overkill to create a new project for each (more on how to decide this below).
> 
> Instead you can use the environments feature, which is a special kind of resource attribute applied
> to the whole payload received by Logfire. This attribute says which environment the payload comes from.
> 
> You can set the environment for your project when calling `logfire.configure`:
> 
> ```py title="main.py"
> import logfire
> 
> logfire.configure(environment='dev')
> 
> logfire.info("Hi there!")
> ```

Reviewer: I think I need to be a bit more annoying. I think we can simplify this, and be more concise.

I don't think the first paragraph introduces well what we are going to see in this page.

```suggestion
As a developer, we find ourselves working on differents environments for a project: local, production, sometimes staging, and depending on your company deployment strategy... You can have even more! 😅 

With **Logfire** you can distinguish which environment you are sending data to.
You just need to set the the [`environment` parameter on the `logfire.configure`][logfire.configure(environment)].

    ```py title="main.py"
    import logfire

    logfire.configure(environment='local')  # (1)!
    ```
1. Usually you are retriving the environment information from an environment variable. 
```

Author: yeah this is better
